Added recipe dir-treeview-themes

tilmanrassy opened the pull request:
### Brief summary of what the package does

Provides themes for dir-treeview, a file manager for Emacs (cf. https://github.com/tilmanrassy/emacs-dir-treeview).

### Direct link to the package repository

https://github.com/tilmanrassy/emacs-dir-treeview-themes

### Your association with the package

I'm the maintainer of the package.

### Relevant communications with the upstream package maintainer

**None needed** 

### Checklist

<!-- Please confirm by replacing `[]` with `[x]`: -->

- [x] The package is released under a [GPL-Compatible Free Software License](https://www.gnu.org/licenses/license-list.en.html#GPLCompatibleLicenses)
- [x] I've read [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [x] I've used the latest version of [package-lint](https://github.com/purcell/package-lint) to check for packaging issues, and addressed its feedback
- [x] My elisp byte-compiles cleanly
- [x] `M-x checkdoc` is happy with my docstrings
- [x] I've built and installed the package using the instructions in [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)

<!-- After submitting, please fix any problems the CI reports. -->

riscy commented:
Thanks for this.  Quick first pass --

`dir-treeview-hortensia-theme.el` with checkdoc 0.6.2:
```
dir-treeview-hortensia-theme.el:0: The first line should be of the form: ";;; package --- Summary"
dir-treeview-hortensia-theme.el:0: You should have a section marked ";;; Commentary:"
dir-treeview-hortensia-theme.el:2: You should have a section marked ";;; Code:"
dir-treeview-hortensia-theme.el:68: The footer should be: (provide 'dir-treeview-hortensia-theme)\n;;; dir-treeview-hortensia-theme.el ends here
```

`dir-treeview-hortensia-theme.el` with package-lint 20221003.1636:
```
2 issues found:
2:0: error: "dir-treeview-hortensia" doesn't start with package's prefix "dir-treeview-themes".
49:5: warning: Closing parens should not be wrapped onto new lines.
```

`dir-treeview-petunia-theme.el` with checkdoc 0.6.2:
```
dir-treeview-petunia-theme.el:0: The first line should be of the form: ";;; package --- Summary"
dir-treeview-petunia-theme.el:0: You should have a section marked ";;; Commentary:"
dir-treeview-petunia-theme.el:2: You should have a section marked ";;; Code:"
dir-treeview-petunia-theme.el:68: The footer should be: (provide 'dir-treeview-petunia-theme)\n;;; dir-treeview-petunia-theme.el ends here
dir-treeview-petunia-theme.el:3: Lisp symbol `dir-treeview' should appear in quotes
```

`dir-treeview-petunia-theme.el` with package-lint 20221003.1636:
```
2 issues found:
2:0: error: "dir-treeview-petunia" doesn't start with package's prefix "dir-treeview-themes".
49:5: warning: Closing parens should not be wrapped onto new lines.
```

`dir-treeview-themes.el` with package-lint 20221003.1636:
```
1 issue found:
1:0: error: There is no (provide 'dir-treeview-themes) form.
```
I think this is fine, albeit a bit unusual.  I also notice the `;;; Commentary` in this file wraps a little widely -- aiming for a traditional ~80 characters would help (e.g. with `M-q`).

Package and license:
- unknown license -- dir-treeview-hortensia-theme.el: (no header)
- Add *formal* license boilerplate or an [SPDX-License-Identifier](https://spdx.dev/ids/) to dir-treeview-hortensia-theme.el
- unknown license -- dir-treeview-petunia-theme.el: (no header)
- Add *formal* license boilerplate or an [SPDX-License-Identifier](https://spdx.dev/ids/) to dir-treeview-petunia-theme.el

tilmanrassy commented:
Hi @riscy ,

thank you for your quick reaction!

I fixed the long lines in dir-treeview-themes.el and added licenses in dir-treeview-hortensia-theme.el and dir-treeview-petunia-theme.el.

Best regards,
Tilman

riscy commented:
Thanks @tilmanrassy -- here are the latest salient issues:

`dir-treeview-hortensia-theme.el` with package-lint 20221003.1636:
```
2 issues found:
28:0: error: "dir-treeview-hortensia" doesn't start with package's prefix "dir-treeview-themes".
75:5: warning: Closing parens should not be wrapped onto new lines.
```

`dir-treeview-petunia-theme.el` with package-lint 20221003.1636:
```
2 issues found:
29:0: error: "dir-treeview-petunia" doesn't start with package's prefix "dir-treeview-themes".
76:5: warning: Closing parens should not be wrapped onto new lines.
```

You could also quiet up the checkdocs by adding one line to the end of the theme files, e.g.
```
;;; dir-treeview-hortensia-theme.el ends here
```

tilmanrassy commented:
Hi @riscy ,

thank you for the feedback!

I was able to get rid of the "Closing parens should not be wrapped onto new lines" warning, and added the line you suggested at the end of the files. I'm left with

```
28:0: error: "dir-treeview-hortensia" doesn't start with package's prefix "dir-treeview-themes".
```
(and the same for the other theme). I'm a bit unsure what do do. I could change
```
(deftheme dir-treeview-hortensia
  "A dir-treeview theme with icons based on the Oxygen icon set.")
```
to
```
(deftheme dir-treeview-themes-hortensia
  "A dir-treeview theme with icons based on the Oxygen icon set.")
```
But shouldn't I change the filename `dir-treeview-hortnesia-theme.el` to `dir-treeview-themes-hortensia-theme.el`, then? I would prefer the shorter filename, which doesn't have two times "theme" in it. In addition, changing the filename requires changes in the code of the main package `dir-treeview`, which I would prefer to avoid.

Anyway, both renaming the files and the code changes are not that problematic, I can do that if it's necessary!

Best regards,
Tilman

riscy commented:
There's [an answer in a rough FAQ](https://github.com/melpa/melpa/blob/4caa9a03cad40d24484fce27fcfcf75b1d08e814/FAQ.md#how-do-i-make-a-package-containing-multiple-themes) I was sketching out, but you're right with what you write - it does become tricky, because you do want the filename to end with `-theme.el`, but you also want them to have the same prefix to help with namespacing issues.

tilmanrassy commented:
Hi @riscy ,

apologies for the delay!

Thank you for pointing me to the FAQ.

I reworked the theme package (dir-treeview-theme) and adapted the main package (dir-treeview).

The package-lint errors are now gone.

A also applied flycheck and checkdoc. There are are few messges left, but no real errors, I think.

Perhaps you can have a look at the flycheck messages in dir-treeview-themes.el. I added a comment in the code explaining why I think the problem flycheck reports does not exist in reality.

Best regards,
Tilman

riscy commented:
Thanks!  Taking a look now --

Re: this comment
```elisp
;; Flycheck complains here, saying that the following functions are not knwon to be defined.
;; However, the functions are defined in dir-treeview, and the progn is only executed when  
;; dir-treeview was loaded, so it's guaranteed that the functions are defined when called.  
```
why not just `(require 'dir-treeview)` and remove the conditional?  dir-treeview is already declared as a dependency, so this seems like the right move to me and it automatically solves the byte-compile nags.

`dir-treeview-themes-petunia-theme.el` with checkdoc 0.6.2:
```
dir-treeview-themes-petunia-theme.el:30: Lisp symbol `dir-treeview' should appear in quotes
```
This one looks easy to fix, and following this lint will create a convenient hyperlink in the help for your users.

tilmanrassy commented:
Hi @riscy ,

I fixed the two issues you mentioned.

Best regards,
Tilman

riscy commented:
Thanks!  Looks good to go.

tilmanrassy commented:
Hello @riscy,
Thank you!
